site/app/views/calendar/CalendarView.php

```diff
@@ -40,11 +40,24 @@ public function showCalendar(CalendarInfo $info, array $courses, bool $in_course
         //Create list of courses and their term and get color
         $formatted_courses = [];
         $unformatted_courses = [];
+        $display_names = [];
         //$course_colors = [];
         foreach ($courses as $course) {
-            $course_string = sprintf("%s %s", $course->getTitle(), $course->getTerm());
+            $course_string = $course->getTitle() . " " . $course->getTerm();
             array_push($formatted_courses, $course_string);
         }
+        $courseWithName = [];
+        foreach ($courses as $index => $course) {
+            $nameString = $course->getDisplayName() ?
+                        ($course->getDisplayName() . " " . $course->getTerm()) :
+                        ($course->getTitle() . " " . $course->getTerm());
```

Comment:
An even cleaner solution you could consider:
```suggestion
            $displayName = $course->getDisplayName()
            $nameString = ($displayName !== '' ? $displayName : $course->getTitle()) . $course->getTerm();
```